src/Explorer/Controls/Tabs/TabComponent.tsx

```diff
@@ -33,11 +36,41 @@ export class TabComponent extends React.Component<TabComponentProps> {
       throw new Error(msg);
     }
   }
+  state = {
```

Comment:
You don't need to define a state for the active tab index. You already have it in the props: `this.props.currrentTabIndex`